Tests/Tests/AESTests.swift

```diff
@@ -473,6 +473,55 @@ extension AESTests {
         XCTAssertEqual(Array(encrypted), [UInt8](hex: "")) // C
         XCTAssertEqual(gcm.authenticationTag, [UInt8](hex: "0xcd33b28ac773f74ba00ed1f312572435")) // T (128-bit)
     }
+
+    func testEncryptTooLongKey() {
+        let key = "f83c08efba542f025f13cfd90110e29a09add0de242b3o587dbc3777b2232dbn".bytes
+        let iv = "0000000000000000".bytes
+
+        do {
+            let aes = try AES(key: key, blockMode: CBC(iv: iv))
+            _ = try aes.encrypt(key)
+            XCTFail("not supposed to success without specified variant")
+        } catch {
+        }
+    }
+
+    func testEncryptTooLongWithVariantKey() {
+        let key = "f83c08efba542f025f13cfd90110e29a09add0de242b3o587dbc3777b2232dbn".bytes
+        let iv = "0000000000000000".bytes
+
+        do {
+            let aes = try AES(key: key, blockMode: CBC(iv: iv), variant: .aes128)
+            _ = try aes.encrypt(key)
+        } catch {
+            XCTFail("an error occured : \(error)")
+        }
+    }
+
+
+    func testEncryptTooShortKey() {
+        let key = "f83".bytes
+        let iv = "0000000000000000".bytes
+
+        do {
+            let aes = try AES(key: key, blockMode: CBC(iv: iv))
+            _ = try aes.encrypt(key)
+            XCTFail("not supposed to success without specified variant")
+        } catch {
+        }
+    }
+
+    func testEncryptTooShortWithVariantKey() {
+        let key = "f83".bytes
+        let iv = "0000000000000000".bytes
+
+        do {
+            let aes = try AES(key: key, blockMode: CBC(iv: iv), variant: .aes128)
+            _ = try aes.encrypt(key)
+        } catch {
+            XCTFail("an error occured : \(error)")
```

Comment:
this is negative test, error is expected, test shouldn't fail in this case

Reply:
why is this test should fail ?

Reply:
the keys is "f83", that should throw error. This PR is all about it, isn't it?

Reply:
The first test tooShort `testEncryptTooShortKey` , is supposed to pass in the catch, so, there is no XCTFailt in there but in the do. 
But the `testEncryptTooShortWithVariantKey ` test use the aes128 variant, so It doesn't use the key length to compute the variant, so it is normal to pass

Reply:
what is the valid AES variant for key "f83" ?

Reply:
I agree with you that the inferred variant is the invalidKey error, but in my unit test, I am not using the inferred one